src/lib/components/collections/tab-bar/TabBar.svelte

```diff
@@ -0,0 +1,127 @@
+<script lang="ts">
+  // ---- SVG
+  import plusIcon from "$lib/assets/actionicon-normal.svg";
+  import angleLeft from "$lib/assets/angleLeft.svg";
+  import angleRight from "$lib/assets/angle-right.svg";
+
+  // ---- Store
+  import { collapsibleState } from "$lib/store/request-response-section";
+
+  // ---- Interface
+  import type { TabDocument } from "$lib/database/app.database";
+  import type { NewTab } from "$lib/utils/interfaces/request.interface";
+
+  // ---- Component
+  import Tab from "../req-res-section/sub-components/sub-components-header/Tab.svelte";
+
+  // ---- Helper
+  import { moveNavigation } from "$lib/utils/helpers/navigation";
+
+  export let tabList: TabDocument[] = [];
+  export let onNewTabRequested: () => void;
+  export let onTabClosed: (id: string, tab: NewTab) => void;
+  export let onDropEvent: (event: Event) => void;
+  export let onDragStart: (index: number) => void;
+  export let onDropOver: (index: number) => void;
+  export let onTabSelected: (id: string) => void;
+
+  $: {
+    if (tabList) {
```

Comment:
Could we find a shortcut or more optimized way to achieve the tab width?

Reply:
I tried to find a more optimized way but it didn't worked like here is one of the approach 
`const tabWidthLookup = {
  0: 100, // Default tab width
  1: 100,
  2: 100,
  3: 100,
  4: 100,
  5: 196,
  6: 140,
  7: 140,
  8: 140,
  9: 140,
  10: 140,
};

if (tabList && tabList.length >= 0) {
  tabWidth = tabWidthLookup[tabList.length] || tabWidthLookup[0];
}
`

but here issue comes is the new length of tablist gets updated but that if condition doesn't run I tried to add $  so the condition runs whenever tab length changes but it was not working. Any better idea?